app/src/main/java/com/nervousfish/nervousfish/activities/VisualVerificationActivity.java

```diff
@@ -59,7 +59,7 @@ private void nextActivity() {
             LOGGER.info("Sending my profile with name: " + myProfile.getName() + ", public key: "
                     + myProfile.getPublicKey().toString());
             final Contact myProfileAsContact = new Contact(myProfile.getName(), new SimpleKey("simplekey", "73890ien"));
-            this.serviceLocator.getBluetoothHandler().send(myProfileAsContact);
+            this.serviceLocator.getBluetoothHandler().send(this.serviceLocator.getBluetoothHandler().objectToBytes(myProfileAsContact));
```

Comment:
> we do not want duplicate code, although both NFC and Bluetooth have to convert a message to bytes

Not sure if you are aware of this, but you're actually introducing duplicate code because now you constantly have to convert these things to bytes separately. I don't know where the other duplicate code would appear since the problem you're describing is based in a single class. Can you explain where we would have had duplicate code if it where not for this change?


> we want a function to have a single task, and send(Serializable) both converts the serializable object to bytes and makes sure that these bytes are sent

Indeed a single task, **send data**. I don't care what data, is it a string? send it. Is it an object? send it. I don't care how it has to be send... Ow it has to be send as a byte-array? Well let the `send()` method handle that.


> I hate it as much as you do, but it seems that it's not possible to the NFCHandler send messages. It has to be done by the current activity. You cannot "send" messages, but you fill a buffer when the activity starts with the data you want to beam to the other activity. Android beams the buffer constantly to the ether. The send() method is therefore completely useless for NFC.

OK, sure. So we need a send method that accepts byte-arrays and one that accepts object? That works for me...